server/core/src/main/java/org/infinispan/server/core/dataconversion/ProtostreamTranscoder.java

```diff
@@ -0,0 +1,146 @@
+package org.infinispan.server.core.dataconversion;
+
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_JSON;
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_OBJECT;
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_OCTET_STREAM;
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_PROTOSTREAM;
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_UNKNOWN;
+import static org.infinispan.commons.dataconversion.MediaType.TEXT_PLAIN;
+import static org.infinispan.commons.dataconversion.StandardConversions.convertTextToObject;
+import static org.infinispan.commons.dataconversion.StandardConversions.decodeOctetStream;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.StringReader;
+import java.util.Optional;
+
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.commons.dataconversion.OneToManyTranscoder;
+import org.infinispan.commons.dataconversion.StandardConversions;
+import org.infinispan.commons.marshall.ProtoStreamMarshaller;
+import org.infinispan.commons.marshall.WrappedByteArray;
+import org.infinispan.commons.util.Util;
+import org.infinispan.protostream.ProtobufUtil;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+/**
+ * Transcode between application/x-protostream and commons formats
+ *
+ * @since 10.0
+ */
+public class ProtostreamTranscoder extends OneToManyTranscoder {
+
+   protected final static Log logger = LogFactory.getLog(JBossMarshallingTranscoder.class, Log.class);
+   private volatile SerializationContext ctx;
+   private final ProtoStreamMarshaller marshaller;
+   private final ClassLoader classLoader;
+
+   public ProtostreamTranscoder(SerializationContext ctx, ClassLoader classLoader) {
+      super(APPLICATION_PROTOSTREAM, APPLICATION_OCTET_STREAM, TEXT_PLAIN, APPLICATION_OBJECT, APPLICATION_JSON, APPLICATION_UNKNOWN);
+      this.ctx = ctx;
+      this.marshaller = new ProtoStreamMarshaller(ctx);
+      this.classLoader = classLoader;
+   }
+
+   public void setCtx(SerializationContext ctx) {
+      this.ctx = ctx;
```

Comment:
`ProtoStreamMarshaller` is also using the `SerializationContext`. Shouldn't be updated?

Reply:
Good point. I'll remove the marshaller and allows use the `SerializationContext`.